core/templates/pages/teach-page/teach-page.component.ts

```diff
@@ -15,63 +15,128 @@
-  activeTabName: string = this.TAB_ID_TEACH;
-
+  isWindowNarrow: boolean = false;
+  classroomUrlFragment: string;
+  classroomUrl :string;
+  displayedTestimonialId: number;
+  testimonialCount: number;
+  testimonials = [];
+  userIsLoggedIn: boolean = null;
   constructor(
     private siteAnalyticsService: SiteAnalyticsService,
     private urlInterpolationService: UrlInterpolationService,
-    private windowRef: WindowRef
+    private windowDimensionService: WindowDimensionsService,
+    private windowRef: WindowRef,
+    private userService: UserService,
+    private loaderService: LoaderService
   ) {}
 
   ngOnInit(): void {
-    const hash = this.windowRef.nativeWindow.location.hash.slice(1);
-    if (this.ALLOWED_TABS.includes(hash)) {
-      this.activeTabName = hash;
-    }
-    this.windowRef.nativeWindow.onhashchange = () => {
-      const hashChange = this.windowRef.nativeWindow.location.hash.slice(1);
-      if (this.ALLOWED_TABS.includes(hashChange)) {
-        this.activeTabName = hashChange;
-      }
-    };
+    this.isWindowNarrow = this.windowDimensionService.isWindowNarrow();
+    this.displayedTestimonialId = 0;
+    // Change count after all testimonials are available.
+    this.testimonialCount = 2;
+    this.testimonials = this.getTestimonials();
+    this.classroomUrl = this.urlInterpolationService.interpolateUrl(
+      '/learn/<classroomUrlFragment>', {
+        classroomUrlFragment: splashConstants.DEFAULT_CLASSROOM_URL_FRAGMENT
+      });
+    this.loaderService.showLoadingScreen('Loading');
+    this.userService.getUserInfoAsync().then((userInfo) => {
+      this.userIsLoggedIn = userInfo.isLoggedIn();
+      this.loaderService.hideLoadingScreen();
+    });
+  }
+  // TODO(#11657): Extract the testimonials code into a separate component.
+  // The 2 functions below are to cycle between values:
+  // 0 to (testimonialCount - 1) for displayedTestimonialId.
+  incrementDisplayedTestimonialId(): void {
+  // This makes sure that incrementing from (testimonialCount - 1)
+  // returns 0 instead of testimonialCount,since we want the testimonials
+  // to cycle through.
+    this.displayedTestimonialId = (
+      this.displayedTestimonialId + 1) % this.testimonialCount;
   }
 
-  onTabClick(tabName: string): Window {
-    // Update URL hash.
-    this.windowRef.nativeWindow.location.hash = '#' + tabName;
-    this.activeTabName = tabName;
-    return this.windowRef.nativeWindow;
+  decrementDisplayedTestimonialId(): void {
+  // This makes sure that decrementing from 0, returns
+  // (testimonialCount - 1) instead of -1, since we want the testimonials
+  // to cycle through.
+    this.displayedTestimonialId = (
+      this.displayedTestimonialId + this.testimonialCount - 1) %
+      this.testimonialCount;
   }
 
-  getStaticImageUrl(imagePath: string): string {
-    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  getTestimonials():[{}, {}] {
```

Comment:
```suggestion
  getTestimonials(): [{}, {}] {
```
Also, I think the type should be more precise, look into other places in the code where we use dict types.

Reply:
done.

Reply:
Not done?

Reply:
![Screenshot from 2021-01-13 09-04-48](https://user-images.githubusercontent.com/68547101/104403457-92ad4000-557e-11eb-9182-4c0faccf651c.png)
![Screenshot from 2021-01-13 09-05-11](https://user-images.githubusercontent.com/68547101/104403470-96d95d80-557e-11eb-8172-49690aa0a515.png)
I have added these.